src/units/abilities.cpp

```diff
@@ -199,11 +199,19 @@ int find_direction(const map_location& loc, const map_location& from_loc, std::s
 }
 
 /**
- * This function return true if locations checked are the same, or if units have same id.
+ * This function return true if @a u not incapacitated and if not *this.
  */
-bool same_unit(const unit& u, const unit& unit)
+bool unit_valid(const unit& u, const std::size_t& underlying_id)
 {
-	return (u.get_location() == unit.get_location() || u.id() == unit.id());
+	return (!u.incapacitated() && u.underlying_id() != underlying_id);
+}
+
+/**
+ * This function return true if u has an ability with [affect_adjacent] and distance with *this not greater what its max radius.
+ */
+bool unit_has_ability_distant(const std::optional<int>& has_ability_distant, const std::size_t& distance)
+{
+	return(has_ability_distant && has_ability_distant <= distance);
```

Comment:
Should this be `distance <= has_ability_distance`?

I'm wondering why `unit::has_ability_distant()` returns `std::optional<std::size_t>` instead of a plain `std::size_t` with zero meaning "no abilities that affect units one or more hexes away".
```
	/**
	 * If this unit has abilities with [affect_adjacent] subtags, returns the radius of the one with the furthest reach.
	 *
	 * If the unit has no such abilities, returns zero.
	 */
	std::size_t max_ability_radius() const
```

Reply:
Note: zero / non-zero will automatically convert to bool as needed.